interop/src/bin/client_new_codegen.rs

```diff
@@ -0,0 +1,169 @@
+/*
```

Comment:
I think we should rename these to `client-prost` and `client-protobuf` etc, thoughts? Or actually another thought can we just merge the client files together and make it an option which protobuf library it uses? That way its really easy to make some sort of test matrix?

Reply:
I made the changes to have a single client binary that chooses to test a prost or protobuf client based on a command line flag. I introduced a trait with all the test scenarios and added implementations for prost and protobuf.